### PR TITLE
test(e2e): add e2e tests for step-input core

### PR DESCRIPTION
**Pull request by maxym-valor**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#

#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [ ] Run npm run build-pack-library and test in external application

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    })
+
+    it('should check entering invalid values in inputs', () => {
+        for (let i = 0; i < sections.length; i++) {
+            checkInputWithInvalidValues(sections[i])
+        }
+    })
+
+    it('should check disabled inputs', () => {
+        const defaultValue = getValue(formExample + input, 2);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:90]: instead, can get element by disabled class and check that it's buttons and input aren't clickable
for (...) {
expect(isClickable(selector, index)).toBe(false);
}
or
expect(checkElArrIsClickable(selector)).toBe(false);

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        expect(getElementClass(formExample + step, 2)).toContain('is-disabled'); 
+        click(formExample + plusButton, 2);
+        expect(getValue(formExample + input, 2)).toEqual(defaultValue);
+
+        expect(getElementClass(formExample + step, 3)).toContain('is-readonly'); 
+        expect(isElementDisplayed(formExample + plusButton, 3)).toBe(false);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:96]: no need to check this here, visual regression will cover it. just check input not clickable

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    it('should check RTL orientation', () => {
+        stepInputPage.checkRtlSwitch();
+    })
+
+    it('should check examples visual regression', () => {
+        stepInputPage.saveExampleBaselineScreenshot();
+        expect(stepInputPage.compareWithBaseline()).toBeLessThan(5);
+    })
+
+    function checkInputWithInvalidValues(section: string) {
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:109]: needs a typeOf def (i.e. :void)

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        let inputLength = getElementArrayLength(section + input);
+        let defaultValue;
+        section == formExample || section == localExample ? inputLength = 2 : ''
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:112]: `===`

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        for (let i = 0; i < inputLength; i++) {
+            defaultValue = getValue(section + input, i)
+            setValue(section + input, 'asd123', i)
+            sendKeys('Enter')
+            expect(getValue(section + input, i)).toEqual(defaultValue)
+        }
+    }
+
+    function checkInputWorking(section: string, way: 'click' | 'type', sign: 'positive' | 'negative') {
```
[the remark this answers is not in the document]
Comment (Seamoo13) [reply e2e/wdio/core/tests/step-input.e2e-spec.ts:121]: also seems like you could make simple methods instead:
checkIncreaseValueBtn() 
- get start value
- click + button
- expect newValue toBeGreaterThan startValue

many simple methods are better than 1 super complex one ;)

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    getValue,
+    doubleClick,
+    pause,
+    getText,
+    clearValue,
+    setValue,
+    getElementClass,
+    sendKeys,
+    isElementDisplayed
+} from '../../driver/wdio';
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:15]: remove unused methods from imports

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        configExample,
+        defaultExample,
+        currencyExample,
+        step,
+        input,
+        plusButton,
+        minusButton,
+        text,
+        textForDisabledExample
+    } = stepInputPage;
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:34]: remove unused imports (defaultExample, labelExample)

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+    it('should check default example inputs', () => {
+        for (let i = 0; i < sections.length; i++) {
+            checkInputWorking(sections[i], 'click', 'positive')
+            checkInputWorking(sections[i], 'click', 'negative')
+            checkInputWorking(sections[i], 'type', 'negative')
+            checkInputWorking(sections[i], 'type', 'positive')
+        }
+    })
+
+    it('should check limitation for step-input', () => {
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:53]: unclear. you say limitation, but you are expecting class to contain disabled, i don't get it

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+        setValue(configExample + input, '10', 5);
+        expect(getElementClass(configExample + plusButton, 5)).toContain('is-disabled')
+        setValue(configExample + input, '-10', 5);
+        expect(getElementClass(configExample + minusButton, 5)).toContain('is-disabled')
+
+    })
+
+    it('should verify that in specific input step is 0.5', () => {
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:62]: I think you could just do setValu to 0 and hit + button and expect the new value to be 0.5
or
get startValue
click + 
expect newValue to be startValue + 0.5

Comment (Seamoo13) [reply e2e/wdio/core/tests/step-input.e2e-spec.ts:62]: should also check the `-` step

File: e2e/wdio/core/tests/step-input.e2e-spec.ts [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+        stepInputPage.saveExampleBaselineScreenshot();
+        expect(stepInputPage.compareWithBaseline()).toBeLessThan(5);
+    })
+
+    function checkInputWithInvalidValues(section: string) {
+        let inputLength = getElementArrayLength(section + input);
+        let defaultValue;
+        section == formExample || section == localExample ? inputLength = 2 : ''
+        for (let i = 0; i < inputLength; i++) {
+            defaultValue = getValue(section + input, i)
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/step-input.e2e-spec.ts:114]: remember to add `;`

**Review by Seamoo13 — CHANGES_REQUESTED** [review]
all browsers failed + additional comments

File: e2e/wdio/core/tests/step-input.e2e-spec.ts (hunk 8 above)
Comment (maxym-valor) [reply e2e/wdio/core/tests/step-input.e2e-spec.ts:53]: in this case maximum value is 10, so if value is 10 - plus button should be disabled. The same with minimum (-10). Anyway i did it more clearly

Comment (Seamoo13) [reply e2e/wdio/core/tests/step-input.e2e-spec.ts:53]: aha, ok, so please rename to something like 'should check +/- button disabled when input limit reached'

also need to add another test case
it('should not allow input above/below max') and add value by sending keys 20, -20 hit enter key and check input field shows 10, -10

**Pull request closed without merging**